driver-core/src/main/com/mongodb/internal/connection/LoadBalancedCluster.java

```diff
@@ -259,6 +259,7 @@ public void close() {
             if (dnsSrvRecordMonitor != null) {
                 dnsSrvRecordMonitor.close();
             }
+            server.close();
```

Comment:
I think we need a null check for server.

Reply:
Thanks!  Indeed, not only we need to account for it being `null`-able, but also now we need it to be `volatile`, because `close()` does not call `waitForSrv()` before reading from `server`.

Reply:
I am sorry, the previous attempt also was incorrect. Hopefully, now it is.